src/Lib.P2P/MultiAddressWhiteList.cs

```diff
@@ -0,0 +1,87 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using MultiFormats;
+
+namespace Lib.P2P
+{
+    /// <summary>
+    ///   A sequence of filters that are approved.
+    /// </summary>
+    /// <remarks>
+    ///   Only targets that are a subset of any filters will pass.  If no filters are defined, then anything
+    ///   passes.
+    /// </remarks>
+    public class MultiAddressWhiteList : ICollection<MultiAddress>, IPolicy<MultiAddress>
+    {
+        private ConcurrentDictionary<MultiAddress, MultiAddress> _filters =
+            new ConcurrentDictionary<MultiAddress, MultiAddress>();
+
+        /// <inheritdoc />
+        public bool IsAllowed(MultiAddress target)
+        {
+            if (_filters.IsEmpty)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)